.github/workflows/on-merge.yml

```diff
@@ -0,0 +1,71 @@
+# This workflow is triggered every time a change is pushed to any branches
+# Github actions command reference: https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions
+name: On merge to master
+
+# The workflow could also be triggered on PRs
+on:
+  push:
+    branches:
+      - 'master'
+    tags-ignore:
+      - '**'
+
+jobs:
+  build:
+    name: Build Components
+    runs-on: self-hosted
+    env:
+      NEXUS_INTERNAL_URL: ${{ secrets.NEXUS_INTERNAL_URL }}
+    container:
+      image: jahia/cimg-mvn-cache:ga_cimg_openjdk_11.0.20-node
+      credentials:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_PASSWORD }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Execute yarn build
+        shell: bash
+        run: |
+          yarn install
+          yarn build
+
+  test:
+    name: Test Components
+    needs: [build]
+    runs-on: self-hosted
+    container:
+      image: jahia/cimg-mvn-cache:ga_cimg_openjdk_11.0.20-node
+      credentials:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_PASSWORD }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install dependencies
+        shell: bash
+        run: |
+          yarn install
+      - name: Run linting
+        shell: bash
+        run: |
+          yarn lint
+      - name: Run tests
+        shell: bash
+        run: |
+          yarn test
+
+  publish:
+    name: Publish Components
+    needs: [test]
+    runs-on: self-hosted
+    container:
+      image: jahia/cimg-mvn-cache:ga_cimg_openjdk_11.0.20-node
+      credentials:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_PASSWORD }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Execute yarn publish
+        shell: bash
+        run: |
+          yarn install
```

Comment:
I think we're missing the publish step here.

What you should probably do is have a test job, that executes the tests and does the lining.

Then a build and publish job if the previous one was successful.

Reply:
The publish step is just below. And there's already a test job.
I've splitted the steps et moved the tests job to be run first